bims/api_views/location_site.py

```diff
@@ -288,6 +282,34 @@ def get(self, request):
         except ValueError:
             return Response(response_data)
 
+    def get_site_occurrences_per_year(self, collection_records):
+        result = {}
+        occurrences_data = []
+        result['occurrences_line_chart'] = {}
+        result['occurrences_line_chart']['values'] = []
+        result['occurrences_line_chart']['keys'] = []
+        result['occurrences_line_chart']['title'] \
+            = 'Occurrences'
+
+        for each_record in collection_records:
```

Comment:
I avoided looping the collection records because there will be a case with hundred thousand data. Please don’t do this if you can.

Reply:
Did they specify the chart like that? Wouldnt it make more sense to have the start year then yearly intervals until the end year. The years that have no data would have a 0 size bar. How did you do the other charts like that @dimasciput ?

Reply:
@timlinux I usually do the chart with database aggregation, so it’s faster. And it only returns year with data.